yeonjae/programmers-1/예산.js

```diff
@@ -0,0 +1,32 @@
+function solution(d, budget) {
+  const departmentBudget = d.sort((a, b) => {
+    return a - b;
+  });
+
+  let answer = 0;
+  let budgetSum = 0;
+
+  for (let i = 0; i < departmentBudget.length; i++) {
+    budgetSum += departmentBudget[i];
+
+    if (budgetSum <= budget) answer++;
+    else break;
+  }
+
+  return answer;
+}
```

Comment:
```suggestion
function solution(d, budget) {
  let answer = 0;
  let budgetSum = 0;
  
  d.sort((a, b) => a - b);

  for (let i = 0; i < d.length; i++) {
    budgetSum += d[i];

    if (budgetSum <= budget) answer++;
    else break;
  }

  return answer;
}
```
- sort는 대표적으로 사이드 이펙트가 있는 배열 메서드입니다. 즉, 원본 배열을 바꿔버립니다. 
- 그리고 굳이 {} 후에 return a - b 할 필요 없이, 화살표 함수는 중괄호 없이 한줄로 작성하면 자동 리턴을 해주기 때문에 간결하게 작성할 수 있습니다 `d.sort((a, b) => a - b);`

```js
const arr = [5, 2 , 3, 4, 1];
arr.sort((a, b) => a - b);

console.log(arr); // [1, 2, 3, 4, 5]
```

Reply:
좋은 피드백 감사합니다 ! 👍🏻👍🏻👍🏻